src/elements/content-sidebar/TaskModal.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+import * as React from 'react';
+import { FormattedMessage } from 'react-intl';
+import BetaFeedbackBadge from '../../features/beta-feedback';
+import Modal from '../../components/modal/Modal';
+import TaskForm from './activity-feed/task-form';
+import messages from '../common/messages';
+import { TASK_TYPE_APPROVAL, TASK_TYPE_GENERAL } from '../../constants';
+
+type TaskModalProps = {
+    error: ?ElementsXhrError,
+    feedbackUrl: string,
+    handleCreateError: Function,
+    handleCreateSuccess: Function,
+    handleModalClose: Function,
+    isTaskFormOpen: boolean,
+    passThrough: Object,
+    taskType: TaskType,
+};
+
+function getMessageForModalTitle(taskType: TaskType): Object {
+    switch (taskType) {
+        case TASK_TYPE_GENERAL:
+            return messages.tasksCreateGeneralTaskFormTitle;
+        case TASK_TYPE_APPROVAL:
+        default:
+            return messages.tasksCreateApprovalTaskFormTitle;
+    }
+}
+
+const focusTargetSelector = '.task-modal input';
+
+const TaskModal = (props: TaskModalProps) => {
+    // CSS selector for first form element
+    // Note: Modal throws an error if this fails to find an element!
+    return (
+        <Modal
+            className="be-modal task-modal"
+            data-testid="create-task-modal"
+            focusElementSelector={focusTargetSelector}
+            isOpen={props.isTaskFormOpen}
```

Comment:
Props should be destructured in the function signature.

Reply:
why do we do this, actually? my impression was it's a stylistic thing, but is there more to it?